src/Strings.f90

```diff
@@ -1254,18 +1254,29 @@ FUNCTION str_to_sik(this,stt,stp) RESULT(i)
   INTEGER(SIK),INTENT(IN),OPTIONAL :: stp
   INTEGER(SIK) :: i
   !
-  CHARACTER(LEN=:),ALLOCATABLE :: scratch
+  INTEGER(SIK) :: begStr,endStr,ioerr
 
-  IF(PRESENT(stt) .OR. PRESENT(stp)) THEN
-    scratch = this%s(stt:stp)
-  ELSE
-    scratch = this%s
+  !Bypass the world if the string isn't allocated
+  ioerr = -HUGE(ioerr)
+  IF(ALLOCATED(this%s)) THEN
+    !Find the beginning
+    IF(PRESENT(stt)) THEN
+      begStr = stt
+    ELSE
+      begStr = 1
+    ENDIF
+    !Find the end
+    IF(PRESENT(stp)) THEN
+      endStr = stp
+    ELSE
+      endStr = LEN(this%s)
+    ENDIF
+    READ(this%s(begStr:endStr),*,IOSTAT=ioerr) i
   ENDIF
 
-  IF(LEN(scratch) > 0) THEN
-    READ(scratch,*) i
-  ELSE
-    i = -HUGE(i)
+  !Test to see if it worked, return 0 on any error code
+  IF(ioerr /= 0) THEN
```

Comment:
So ioerr would be the normal error codes returned by the READ statement. Which I only returns 0 if there was no error. I am not convinced what codes GCC returns when it encounters an error, but I also am not really convinced it matters in this case, it could matter if we were trying to anticipate some common errors, but that feels scary.

I also don't really care for returning zero, I just wasn't really sure what other number to pick. Too bad there isn't an integer NaN that would be ideal. I suppose `-HUGE(i)` is probably the the best bu it is also valid input albeit not one we should really use often.

Reply:
Yeah... I would prefer `NaN`, then `-Infinity`, and then `-HUGE(i)`... I'm guessing neither of the first 2 exist for integers, but `-HUGE(i)` is way less likely to cause a problem than `0`

Reply:
fixed